tests/Controller/HelperControllerTest.php

```diff
@@ -312,6 +316,67 @@ public function testsetObjectFieldValueAction()
         $this->assertEquals(200, $response->getStatusCode());
     }
 
+    public function testsetObjectFieldValueActionOnARelationField()
+    {
+        $object = new AdminControllerHelper_Foo();
+        $associationObject = new AdminControllerHelper_Bar();
+        $request = new Request([
+            'code' => 'sonata.post.admin',
+            'objectId' => 42,
+            'field' => 'bar',
+            'value' => 1,
+            'context' => 'list',
+        ], [], [], [], [], ['REQUEST_METHOD' => 'POST', 'HTTP_X_REQUESTED_WITH' => 'XMLHttpRequest']);
+
+        $fieldDescription = $this->prophesize(FieldDescriptionInterface::class);
```

Comment:
Using `prophecy` here breaks https://github.com/sonata-project/SonataAdminBundle/blob/3.x/CONTRIBUTING.md#using-test-doubles `2.` and `3.`?

Reply:
Wow, it breaks indeed. Do you want me to change all to go back to the shitty createMock? That would be insane..

Reply:
I thought on refactoring the whole class, but it is also not an easy task.

Reply:
Maybe we should rethink that rule?

Reply:
I will work on a full refactor of HelperControllerTest, but on another PR, because it will take some time